demos/jobs/jobs_demo_mosquitto/jobs_demo_mosquitto.c

```diff
@@ -0,0 +1,1264 @@
+/*
+ * AWS IoT Device SDK for Embedded C
+ * Copyright (C) 2020 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/*
+ * This demonstration downloads files from URLs received via AWS IoT Jobs.
+ * Details are available in the usage function below.  Note: This demo
+ * focuses on use of the jobs library; a thorough explanation of libmosquitto
+ * is beyond the scope of the demo.
+ */
+
+#include <assert.h>
+#include <err.h>
+#include <errno.h>
+#include <getopt.h>
+#include <signal.h>
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <sys/wait.h>
+#include <time.h>
+#include <unistd.h>
+
+#include <mosquitto.h>
+#if ( LIBMOSQUITTO_VERSION_NUMBER < 1004010 )
+    #error Please use libmosquitto at version 1.4.10 or higher.
+#endif
+
+#include "demo_config.h"
+#include "jobs.h"
+#include "core_json.h"
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief MQTT server port number
+ *
+ * AWS IoT Core uses this port for MQTT over TLS.
+ */
+#define DEFAULT_MQTT_PORT       ( 8883 )
```

Comment:
We have been exposing the MQTT_PORT config in the `demo_config.h` files for the MQTT demos. It may be good to expose it their along with defining a default here

Reply:
Here, the port can be changed with an argument.  I don't think we need a further configuration for it.